window: Add optional tile size cycling

vkareh opened the pull request:
Adding a new option to allow tile size cycling. When enabled, using the
keyboard shortcut for tiling multiple times in a row cycles the window
through different sizes (1/2 -> 1/3 -> 1/4 -> 3/4 -> 2/3 -> Untiled).

raveit65 commented:
Amazing feature :)
I think first tiling windows position should be `META_TILE_CYCLE_50`, or not?
Otherwise it is confusing when you expect quarter-tililing-left-down but window size is bigger.
Order should be `META_TILE_CYCLE_50` --> `META_TILE_CYCLE_66` --> `META_TILE_CYCLE_33` --> `META_TILE_CYCLE_NONE`
or
 `META_TILE_CYCLE_50` --> `META_TILE_CYCLE_33` --> `META_TILE_CYCLE_66` --> `META_TILE_CYCLE_NONE`.

raveit65 commented:
I forgot.......
It should be the default, IHMO :)

raveit65 commented:
After using it the first hour today it is clear that first window size when starting the cycle have to be `META_TILE_CYCLE_50` or `1/2` size, everything else is confusing ;) 
But it works very nice.

vkareh commented:
Thanks for testing and for your suggestions. I've updated it so that it always starts with 50% of the screen, then goes smaller from there and back to larger, and it feels a lot better. Also I've added 2 extra sizes.

vkareh commented:
> I forgot.......
> It should be the default, IHMO :)

Done

vkareh commented:
@lukefromdc thanks for testing. The tile size choices are also influenced by Ubuntu's Compiz, which has the cycle: 50->33->25->100->75->66. In testing different configurations and orders with @raveit65, we decided to drop the 100 (we have maximize for that), but keep the others, and add the Untiled option to the list (since it follows marco's current behavior, where keyboard tiling toggles between 50 and Untiled).

lukefromdc commented:
Oh yeah: I was comparing to compiz-reloaded, which I have and work on
